cli/parse_args.ts

```diff
@@ -660,50 +663,51 @@ export function parseArgs<
       const letters = arg.slice(1, -1).split("");
 
       let broken = false;
-      for (let j = 0; j < letters.length; j++) {
+      for (const [j, letter] of letters.entries()) {
         const next = arg.slice(j + 2);
 
         if (next === "-") {
-          setArg(letters[j], next, arg);
+          setArg(letter, next, arg);
           continue;
         }
 
-        if (/[A-Za-z]/.test(letters[j]) && /=/.test(next)) {
-          setArg(letters[j], next.split(/=(.+)/)[1], arg);
+        if (/[A-Za-z]/.test(letter) && /=/.test(next)) {
+          setArg(letter, next.split(/=(.+)/)[1], arg);
           broken = true;
           break;
         }
 
         if (
-          /[A-Za-z]/.test(letters[j]) &&
+          /[A-Za-z]/.test(letter) &&
           /-?\d+(\.\d*)?(e-?\d+)?$/.test(next)
         ) {
-          setArg(letters[j], next, arg);
+          setArg(letter, next, arg);
           broken = true;
           break;
         }
 
-        if (letters[j + 1] && letters[j + 1].match(/\W/)) {
-          setArg(letters[j], arg.slice(j + 2), arg);
+        if (letters[j + 1]?.match(/\W/)) {
+          setArg(letter, arg.slice(j + 2), arg);
           broken = true;
           break;
         } else {
-          setArg(letters[j], get(flags.strings, letters[j]) ? "" : true, arg);
+          setArg(letter, get(flags.strings, letter) ? "" : true, arg);
         }
       }
 
-      const [key] = arg.slice(-1);
+      const key = arg.at(-1)!;
```

Comment:
This code is run inside the check `/^-[^-]+/.test(arg)` which means arg needs to be a non-empty string.